inbox/events/microsoft/events_provider.py

```diff
@@ -216,6 +217,12 @@ def webhook_notifications_enabled(self, account: Account) -> bool:
         * https://learn.microsoft.com/en-us/answers/questions/417261/error-on-adding-subscription-on-events-using-ms-gr.html
         * https://stackoverflow.com/questions/65030751/ms-graph-adding-subscription-returns-extensionerror-and-serviceunavailable
         """
+
+        # First check if we already have cached value since this function is called
+        # repeatedly and there is no need to do extra HTTP request every time.
+        if self._webhook_notifications_enabled is not None:
+            return self._webhook_notifications_enabled
```

Comment:
Is it possible that this method is called one time for Account A and another time for Account B? If so, this will have issues. If not, should this method not accept the `account` argument?

Reply:
No. It's always the same account object, I know it's weird given you receive account as parameter but that's how it was done in the past for some reason.

It's:

- single greenlet that is responsible for account sync 
- that has single events provider for this specific account
- that has single client for this specific account

Reply:
I guess this was done like that with Google because they did not want to make repetitive ORM calls in this component, see [Google one](https://github.com/closeio/sync-engine/blob/b606e71f04625e591ffe5220e80fae38d8dda9de/inbox/events/google.py#L287-L409).

Reply:
But I guess the whole thing could be refactored to accept `Account` instance in the constructor.

Reply:
Yea, that would make for sense, I think. Should probably note it as an issue or maybe at least leave a TODO comment.

Reply:
See: https://github.com/closeio/sync-engine/pull/459